### PR TITLE
[draft] Register machine - Add tmps in frame and compiler. Add RETURN_VALUE_R.

### PR DESCRIPTION
**Pull request by iritkatriel**
* add tmps to the frame, between the fast locals and the stack.
* add plumbing in the compiler for a 1-arg reg machine, with RETURN_VALUE_R that uses a tmp register.

### CODE REVIEW

File: Python/compile.c [hunk 1]
```diff
@@ -184,31 +185,71 @@ static struct jump_target_label_ NO_LABEL = {-1};
...
+#define EXPLICIT_OPARG(V) ((const oparg_t){.value=(V), .type=OPARG_TYPE_EXPLICIT})
+#define CONST_OPARG(V)    ((const oparg_t){.value=(V), .type=OPARG_TYPE_CONST})
+#define NAME_OPARG(V)     ((const oparg_t){.value=(V), .type=OPARG_TYPE_NAME})
+#define TMP_OPARG(V)      ((const oparg_t){.value=(V), .type=OPARG_TYPE_TMP})
+
+#define IS_UNUSED(OPARG) ((OPARG).type == OPARG_TYPE_UNUSED)
+#define SAME_REGISTER(R1, R2) (((R1).type == (R2).type) && ((R1).value == (R2).value))
+
+typedef struct instr_opargs_ {
+    oparg_t arg1;
```
Comment (iritkatriel) [inline Python/compile.c:212]: Once we have variable-length instructions, arg2, arg3 should be added here. Then search this file for arg1 and update all those places to work with the other args as well.

File: Python/ceval_macros.h [hunk 2]
```diff
@@ -144,7 +144,8 @@ GETITEM(PyObject *v, Py_ssize_t i) {
 #define NEXTOPARG()  do { \
         _Py_CODEUNIT word = *next_instr; \
         opcode = _Py_OPCODE(word); \
-        oparg1 = oparg = _Py_OPARG(word); \
+        oparg = _Py_OPARG(word); \
+        oparg1 = oparg; \
```
Comment (gvanrossum) [inline Python/ceval_macros.h:148]: I'm not excited about having two variables. In my own 'long-insts' branch I just count `oparg`, `oparg2`, `oparg3`, ...

Comment (iritkatriel) [reply Python/ceval_macros.h:148]: I think the generator currently emits REG(oparg1). We could change that.

Comment (iritkatriel) [reply Python/ceval_macros.h:148]: Or we could "#define oparg1 oparg".

Comment (gvanrossum) [reply Python/ceval_macros.h:148]: I prefer modifying the generator over the macro. REG(oparg1) only occurs once in the generator, explicitly, so it's very easy to change (in fact in ly 'long-instrs' I already changed it.)

Comment (gvanrossum) [reply Python/ceval_macros.h:148]: You can even cherry-pick 8d06d1b1e5 from that branch.

**Pull request closed without merging**